cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/SecureStoreHandler.java

```diff
@@ -0,0 +1,114 @@
+import java.util.List;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * Exposes REST APIs for {@link co.cask.cdap.api.security.store.SecureStore} and
+ * {@link co.cask.cdap.api.security.store.SecureStoreManager}.
+ */
+@Path(Constants.Gateway.API_VERSION_3 + "/security/store/namespaces/{namespace-id}")
+public class SecureStoreHandler extends AbstractAppFabricHttpHandler {
+
+  private final SecureStore secureStore;
+  private final SecureStoreManager secureStoreManager;
+
+  @Inject
+  public SecureStoreHandler(SecureStore secureStore, SecureStoreManager secureStoreManager) {
+    this.secureStore = secureStore;
+    this.secureStoreManager = secureStoreManager;
+  }
+
+  @Path("/key")
+  @PUT
+  public void create(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("namespace-id") String namespace)
+    throws BadRequestException, IOException {
+    SecureStoreCreateRequest secureStoreCreateRequest = parseBody(httpRequest, SecureStoreCreateRequest.class);
+
+    if (secureStoreCreateRequest == null) {
+      throw new BadRequestException("Unable to parse the request.");
+    }
+
+    String name = secureStoreCreateRequest.getName();
+    String description = secureStoreCreateRequest.getDescription();
+    String value = secureStoreCreateRequest.getData();
+    if (name == null || name.isEmpty()) {
+      throw new BadRequestException("Name can not be empty.");
+    }
+    if (value == null || value.isEmpty()) {
+      throw new BadRequestException("Data can not be empty.");
+    }
+    byte[] data = value.getBytes(StandardCharsets.UTF_8);
+    secureStoreManager.put(namespace, name, data, description, secureStoreCreateRequest.getProperties());
+    httpResponder.sendStatus(HttpResponseStatus.OK);
+  }
+
+  @Path("/keys/{key-name}")
+  @DELETE
+  public void delete(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("key-name") String name,
+                     @PathParam("namespace-id") String namespace)
+    throws IOException {
+    secureStoreManager.delete(namespace, name);
+    httpResponder.sendStatus(HttpResponseStatus.OK);
+  }
+
+  @Path("/keys/{key-name}")
+  @GET
+  public void get(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("key-name") String name,
+                  @PathParam("namespace-id") String namespace)
+    throws IOException {
+    SecureStoreData secureStoreData = secureStore.get(namespace, name);
+    String data = new String(secureStoreData.get(), StandardCharsets.UTF_8);
+    httpResponder.sendJson(HttpResponseStatus.OK, data);
+  }
+
+  @Path("/keys")
+  @GET
+  public void list(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("namespace-id") String namespace)
+    throws IOException {
+    List<SecureStoreMetadata> metadataList = secureStore.list(namespace);
+    List<FileSecureStore.SecureStoreListEntry> returnList = new ArrayList<>(metadataList.size());
```

Comment:
Since ``FileSecureStore.SecureStoreListEntry`` is going to be the output of your REST API, you should make it a public class in ``cdap-proto``, so clients like ``cdap-clients``, ``cdap-cli``, ``cdap-integration-tests`` can easily deserialize the output.